src/lib/openjp2/jp2.c

```diff
@@ -1455,6 +1474,63 @@ static OPJ_BOOL opj_jp2_read_cdef(opj_jp2_t * jp2,
     return OPJ_TRUE;
 }
 
+
+static OPJ_BOOL opj_jp2_read_pxfm(opj_jp2_t * jp2,
+                                  OPJ_BYTE * p_pxfm_header_data,
+                                  OPJ_UINT32 p_pxfm_header_size,
+                                  opj_event_mgr_t * p_manager
+                                 )
+{
+    opj_jp2_cdef_info_t *cdef_info;
+    OPJ_UINT16 i;
+    OPJ_UINT32 l_value;
+    OPJ_UINT16 num_channel;
+    OPJ_UINT8 Channel_index;
+
+    assert(jp2 != 00);
+    assert(p_pxfm_header_data != 00);
+    assert(p_manager != 00);
+    (void)p_pxfm_header_size;
+
+    opj_read_bytes(p_pxfm_header_data, &l_value,
+                   2);
+
+    num_channel = (OPJ_UINT16) l_value;
+
+    p_pxfm_header_data += 2;
+
+    if (jp2->numcomps != num_channel) {
+        opj_event_msg(p_manager, EVT_ERROR,
+                      "Mismatch between num comps and PXFM number of channel \n");
+        return OPJ_FALSE;
+    }
+    jp2->pixel_format = (opj_jp2_pixel_format_t*) opj_malloc(jp2->numcomps * sizeof(
+                            opj_jp2_pixel_format_t));
+
+    for (i = 0; i < jp2->numcomps; ++i) {
+        opj_read_bytes(p_pxfm_header_data, &l_value,
+                       2);
+        p_pxfm_header_data += 2;
+        Channel_index = (OPJ_UINT16) l_value;
+        opj_read_bytes(p_pxfm_header_data, &l_value,
+                       2);
+        p_pxfm_header_data += 2;
+        if (Channel_index < jp2->numcomps) {
+            jp2->pixel_format[Channel_index].pixel_format_type = ((OPJ_UINT16) l_value) &
+                    0xF000;
+
+            if (jp2->pixel_format[Channel_index].pixel_format_type == 0 ||
```

Comment:
given the above masking, it is not possible to have pixel_format_type being 0 or 4...